### PR TITLE
2 ➡️ 3

### PR DESCRIPTION
**Pull request by chapulina**
# ➡️ Forward port

Port `ign-cmake2` to `main`

Branch comparison: https://github.com/ignitionrobotics/ign-cmake/compare/main...ign-cmake2

**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)

### CODE REVIEW

File: examples/core_child_private/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.10.2 FATAL_ERROR)
+project(ignition-core_child_private VERSION 0.1.0)
+find_package(ignition-cmake2 REQUIRED)
```
Comment (scpeters) [inline examples/core_child_private/CMakeLists.txt:3]: `ignition-cmake2` -> `ignition-cmake3`

File: tutorials/developing_with_ign-cmake.md [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+The available options are:
+
+* `RelWithDebInfo`: Mostly optimized build, but with debug symbols.
+* `Debug`: Debug build without optimizations, with debug symbols enabled.
+* `Release`: Fully optimized build, no debug symbols.
+* `MinSizeRel`: Fully optimized build, minimal binary size
+* `Coverage`: Build with additional information required for the [`gcov` analysis tool](https://en.wikipedia.org/wiki/Gcov)
+* `Profile`: Use flags that are helpful with the [`gprof` profiling tool](https://en.wikipedia.org/wiki/Gprof)
+
+More information about flags applied can be found in [IgnSetCompilerFlags.cmake](https://github.com/ignitionrobotics/ign-cmake/blob/ign-cmake2/cmake/IgnSetCompilerFlags.cmake)
```
Comment (scpeters) [inline tutorials/developing_with_ign-cmake.md:29]: this link points to the `ign-cmake2` branch

**Review by scpeters — COMMENTED** [review]
tests are failing

**Comment by adlarkin** [comment]
There were a few PRs that have been merged to `ign-cmake2` since this forward port was opened, can you update this PR so that we can forward port these new PRs as well? I believe we are only missing #150, #158, and #159.

**Comment by chapulina** [comment]
> can you update this PR so that we can forward port these new PRs as well?

Done :+1:

**Review by adlarkin — APPROVED** [review]
Looks good to me. We should forward-port more often so that these PRs are smaller :sweat_smile: 

I think we will need to manually set DCO to pass because of the bitbucket PR signoffs.

**Pull request merged**